### PR TITLE
400ZR test for monitoring pre FEC BER.

### PR DESCRIPTION
**Pull request by ahsaanyousaf**

### CODE REVIEW

File: feature/platform/transceiver/ZR_pre-fec_ber_test/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+*   Verify that the optics pre-FEC BER is updated after the interface is
+    flapped.
+
+    *   Enable a pair of ZR interfaces on the DUT as explained above.
+    *   Verify the ZR optics pre FEC BER PMs are in the normal range.
+    *   Disable or shut down the interface on the DUT.
+    *   Re-enable the interfaces on the DUT.
+    *   Verify the ZR optics pre FEC PM is updated to the value in the normal
+        range again.
```
Comment (dplore) [inline feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:38]: should we specify the normal range that is expected to pass the test?

File: feature/platform/transceiver/ZR_pre-fec_ber_test/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+*   Connect two ZR interfaces using a duplex LC fiber jumper such that TX
+    output power of one is the RX input power of the other module.
+*   To establish a point to point ZR link ensure the following:
+      * Both transceivers state is enabled
+      * Both transceivers are set to a valid target TX output power
+        example -10 dBm
+      * Both transceivers are tuned to a valid centre frequency
+        example 193.1 THz
+*   With the link ZR link established as explained above, verify that the
```
Comment (dplore) [inline feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:17]: Is there any expected range of values?  Or just verify that the path exists and any value is returned?

**Review by dplore — CHANGES_REQUESTED** [review]

File: feature/platform/transceiver/ZR_pre-fec_ber_test/README.md (hunk 1 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:38]: Specified that typical expected value for pre-FEC BER should be less than 1.2E-2

File: feature/platform/transceiver/ZR_pre-fec_ber_test/README.md (hunk 2 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:17]: For the TX output power and centre frequency, we need to set these to fix values as called out.

File: feature/platform/transceiver/ZR_pre-fec_ber_test/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    following ZR transceiver telemetry paths exist and are streamed for both
+    the ZR optics
+    *   /terminal-device/logical-channels/channel/otn/state/pre-fec-ber/instant
+    *   /terminal-device/logical-channels/channel/otn/state/pre-fec-ber/avg
+    *   /terminal-device/logical-channels/channel/otn/state/pre-fec-ber/min
+    *   /terminal-device/logical-channels/channel/otn/state/pre-fec-ber/max
+
+**Note:** For min max avg values, performance monitoring bin of 10 sec is
+          preferred. These values should be retrieved using a gNMI.Subscribe
```
Comment (ejbrever) [inline feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:26]: Maybe we can say:

For min, max, and avg values, 10 second sampling is preferred. If 10 seconds is not supported, the sampling interval used must be communicated.

Comment (ejbrever) [inline feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:26]: s/should/must :)

These values must be retrieved using...

File: feature/platform/transceiver/ZR_pre-fec_ber_test/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          with a SAMPLE subscription.
+
+
+*   Verify that the optics pre-FEC BER is updated after the interface is
```
Comment (ejbrever) [inline feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:30]: grammar nit

 Verify that the optics pre-FEC BER is updated after the interface flaps.

File: feature/platform/transceiver/ZR_pre-fec_ber_test/README.md (hunk 3 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:26]: Updated

Comment (ahsaanyousaf) [reply feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:26]: updated :)

File: feature/platform/transceiver/ZR_pre-fec_ber_test/README.md (hunk 4 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/ZR_pre-fec_ber_test/README.md:30]: Done

**Review by dplore — APPROVED** [review]

**Pull request merged**